src/include/homestore/replication_service.hpp

```diff
@@ -52,6 +61,14 @@ class ReplicationService {
     virtual hs_stats get_cap_stats() const = 0;
 
     virtual meta_sub_type get_meta_blk_name() const = 0;
+
+    /// @brief get the leader replica_id of given group
+    virtual const replica_id_t get_leader_id(group_id_t group_id) const = 0;
+
+    /// @brief get replication status. If called on follower member
+    /// this API can return empty result.
+    virtual std::vector<peer_info> get_replication_status(group_id_t group_id) const = 0;
+
```

Comment:
Is there a use case where we have this api in ReplicationService or can this be at every ReplDev itself, because we have ReplDev::is_leader() already and we could do ReplDev::get_leader() and get_replication_status() there right?

Reply:
Yes there is no strong requirement to have that in ReplicationService as in ReplicationService it also get the ReplDev and operate based on that.. was trying to have a wrapper so that client(external HS) doesnt need to get ReplDev to get the stats when they want to report metrics.  However in HO we already keep the ReplDev ptr in HS_PG.

Will move those to ReplDev.